### ISSUE TITLE
gzip HTML responses

### ISSUE THREAD

**Issue by kevinrobinson** [studentinsights/studentinsights#1196]
We aren't doing this now for the HTML views that Rails renders (or for JSON like the names endpoint).  We should so that we send less bytes and pages load faster (assuming the CPU impact is low).  Looks like this might be as simple as dropping in a gem: https://github.com/romanbsd/heroku-deflater

**Comment by kevinrobinson** [comment]
This is an example of what Web Inspector's performance audit tool suggests might be the gains from doing this - an 83% reduction in bytes:

<img width="696" alt="screen shot 2017-10-25 at 1 22 50 am" src="https://user-images.githubusercontent.com/1056957/31981732-18410acc-b923-11e7-8878-592771629a8a.png">

**Comment by kevinrobinson** [comment]
Adding `config.middleware.use Rack::Deflater` works on JSON responses, but doesn't seem to reduce the bytes transferred for HTML responses (in fact increases it by a tiny amount).  The `Rack::Deflater` middleware does work on JSON responses, and the names endpoint shrinks to ~67% of its size.

To sanity check that gzip would reduce the file size for this kind of data, I ran gzip on the command line for that HTML response and it compresses it to ~10% of the size, and with a production overview HTML file it cuts it to 9% (!) of the size.  Rails has `ActiveSupport::Gzip`, which also gets similarly great compression when run on that plain HTML file in a Rails console.  

So it seems like there must be something particular about the middleware and HTML responses.  I added logs into Rack::Deflater and confirmed that the gzip path [here](https://github.com/rack/rack/blob/master/lib/rack/deflater.rb#L64) executes, and can verify the response headers indicate gzipping on the HTML response (even though the size isn't any smaller).  
  I can also see that the middleware ultimately calls Zlib::GzipWriter, the same class that ActiveSupport::Gzip calls.  The arguments are slightly different, but changing `Rack::Deflater` to match the `ActiveSupport::Gzip` default arguments still doesn't reduce the number of bytes.

So I'm stuck here, feels like I'm missing something but can't see what.  Other options altogether could be: reworking the deployment setup to put nginx in front of Rails (against the grain for Heroku deploys, risky) or splitting out the data from HTML responses and requesting them as JSON endpoints (gets compression at the cost of a second request, probably a good idea longer-term anyway).
